### ISSUE TITLE
DOC: Improve the `random.Generator.shuffle` docs

### ISSUE THREAD

**Issue by tcrasset** [numpy/numpy#21034]
### Issue with current documentation:

Hello!
Currently, when viewing the documentation of the [`random.Generator.shuffle`](https://numpy.org/devdocs/reference/random/generated/numpy.random.Generator.shuffle.html#numpy.random.Generator.shuffle) method, it doesn't explicitly point out the subtle difference in how it [handles the `axis` parameter](https://numpy.org/doc/stable/reference/random/generator.html?highlight=generator#handling-the-axis-parameter) compared to [`random.Generator.permuted`](https://numpy.org/devdocs/reference/random/generated/numpy.random.Generator.permuted.html) . 

`Generator.permuted` has some decent information, and it links back to `Generator.shuffle` in the `See also` section.
However, `Generator.shuffle` does not link back to `Generator.permuted` and has no `See also` section.

In my opinion, when people think about randomly changing the order of an array, they are more likely to search for `shuffle` than `permutation` or `permuted`, and so I think it's important to have as much detail in `shuffle` than in the others.

### Idea or request for content:

I see three improvements.

- From both the `permuted` docs and the `shuffle` docs, link back to how they both [handle the `axis` parameter](https://numpy.org/doc/stable/reference/random/generator.html?highlight=generator#handling-the-axis-parameter).

- Add a `See also` section in `shuffle` to link to `permuted` and `permutation`.

- Show the array before shuffling it in the `shuffle` example, as it's hard to see how it differs from the original `arr`.

Instead of the example :
```
>>> arr = np.arange(9).reshape((3, 3))
>>> rng.shuffle(arr)

>>> arr
array([[3, 4, 5], # random
       [6, 7, 8],
       [0, 1, 2]])
```

do 
```
>>> arr = np.arange(9).reshape((3, 3))
>>> arr
array([[0, 1, 2],
       [3, 4, 5],
       [6, 7, 8]])
>>> rng.shuffle(arr)

>>> arr
array([[3, 4, 5], # random
       [6, 7, 8],
       [0, 1, 2]])
```

and the same for `axis=1`.

Have a great day.

**Comment by bashtage** [comment]
These all seem like good ideas.  Do you want to submit a PR?

**Comment by tcrasset** [comment]
> These all seem like good ideas. Do you want to submit a PR?

@bashtage I can do that, sure!

**Comment by AnkitKumar2698** [comment]
can i do that if you all say 
let me know sir

**Comment by melissawm** [comment]
@AnkitKumar2698 NumPy doesn't assign issues but it looks like @tcrasset plans on working on this, so maybe look for other issues. You do not need to ask for permission to work on any of them.

**Comment by rossbar** [comment]
Closed by #21269
